tartiflette/sdl/schema.py

```diff
@@ -0,0 +1,285 @@
+                        # automatically at import time.
+                        raise ValueError(
+                            "Field `{}` in object `{}` is invalid. "
+                            "The given type `{}` does not exist !".format(
+                                field_name, obj_name, field.gql_type
+                            )
+                        )
+        return True
+
+    def _validate_object_follow_interfaces(self):
+        for obj_name, gtype in self._gql_types.items():
+            if isinstance(gtype, GraphQLObjectTypeDefinition):
+                if gtype.interfaces:
+                    for interface_namedtype in gtype.interfaces:
+                        interface = self._gql_types.get(
+                            interface_namedtype.name
+                        )
+                        if isinstance(
+                            interface, GraphQLInterfaceTypeDefinition
+                        ):
+                            for ifa_field_name, ifa_field in interface.fields.items(
+                            ):
+                                if ifa_field_name in gtype.fields:
+                                    # compare type declaration
+                                    obj_field = gtype.fields.get(
+                                        ifa_field_name
+                                    )
+                                    if obj_field.gql_type != ifa_field.gql_type:
+                                        raise ValueError(
+                                            "Field `{}` in object `{}` "
+                                            "is invalid. The interface "
+                                            "`{}` constraints the field to a "
+                                            "`{}` type, got `{}`".format(
+                                                ifa_field_name, obj_name,
+                                                interface.name,
+                                                ifa_field.gql_type,
+                                                obj_field.gql_type
+                                            )
+                                        )
+                                else:
+                                    raise ValueError(
+                                        "Field `{}` in object `{}` "
+                                        "is missing. The interface `{}` "
+                                        "expects a `{}` field of type "
+                                        "`{}`".format(
+                                            ifa_field_name, obj_name,
+                                            interface.name, ifa_field.name,
+                                            ifa_field.gql_type
+                                        )
+                                    )
+        return True
+
+    def _validate_schema_root_types_exist(self):
+        # Check query, this is mandatory
```

Comment:
Put "query" into parentheses to highlight the fact that Query is a type and not a word in the sentence.

Reply:
Done.